9.7/scripts.js

```diff
@@ -0,0 +1,157 @@
+                newGameBtn.innerHTML = 'Jeszcze raz';
+
+         case 'notStarted':
+        default:        
+                newGameElem.style.display = 'block';
+                pickElem.style.display = 'none';
+                resultsElem.style.display = 'none';
+    }
+}
+
+setGameElements();
+
+//start
+var playerPointsElem    = document.getElementById('js-playerPoints'),
+    playerNameElem      = document.getElementById('js-playerName'),
+    computerPointsElem  = document.getElementById('js-computerPoints'); 
+
+function newGame() {
+    player.name = prompt('Please enter your name', 'imię gracza');
+    if (player.name) {
+        player.score = computer.score = 0;
+        gameState = 'started';
+        setGameElements();
+
+        playerNameElem.innerHTML = player.name;
+
+        setGamePoints();
+
+    }
+}    
+
+//wybór gracza
+function playerPick(playerPick) {
+
+	var computerPick = getComputerPick();
+
+
+	playerPickElem.innerHTML = playerPick;   //wybór gracza
+	 
+	computerPickElem.innerHTML = computerPick;  //wybór komputera
+
+	checkRoundWinner(playerPick, computerPick);
+	setGamePoints();
+
+	checkEndGame(); 
+
+}
+
+//losowanie wyboru komputera
+function getComputerPick() {
+    var possiblePicks = ['rock', 'paper', 'scissors'];
+    return possiblePicks[Math.floor(Math.random() * 3)];
+}
+
+//wyświetlenie wyboru gracza i komputera
+var playerPickElem = document.getElementById('js-playerPick'),
+    computerPickElem = document.getElementById('js-computerPick'),
+    playerResultElem = document.getElementById('js-playerResult'),
+    computerResultElem = document.getElementById('js-computerResult');
+
+//logika gry i przyznawanie punktów
+
+function checkRoundWinner(playerPick, computerPick) {
+    playerResultElem.innerHTML = computerResultElem.innerHTML = ''; 
+
+    var winnerIs = 'player';
+
+    if (playerPick == computerPick) {
+        winnerIs = 'noone';   //remis
+    } else if (
+            (computerPick == 'rock' && playerPick == 'scissors') ||
+            (computerPick == 'scissors' && playerPick == 'paper') ||
+            (computerPick == 'paper' && playerPick == 'rock')) {
+    
+        winnerIs = 'computer';
+    }
+
+    if (winnerIs == 'player') {
+        playerResultElem.innerHTML = 'Win!';
+        player.score++;
+    } else if (winnerIs == 'computer') {
+        computerResultElem.innerHTML = 'Win!';
+        computer.score++;
+    }
+}
+
+//aktualizacja wyniku
+function setGamePoints() {
+
+    playerPointsElem.innerHTML = player.score;
+    computerPointsElem.innerHTML = computer.score;
+}
+
+//koniec gry
+
+function checkEndGame() {
+
+    console.log('endGame');
+
+        if(player.score === 10) {
+        	alert('Gratulacje ' + player.name + '! Wygrałeś');
+
+            gameState = "ended";    
+        }
+        
+        if(computer.score ===10) {
+        	alert('Przegrałeś, spróbuj ponownie');
+
+       		gameState = "ended";
+    
+        }
+ 		setGameElements();
```

Comment:
zauwaz, ze ustawiasz to `gameState = "ended";` dla kazdego warunku ifa tak samo, wiec mozna by przeniesc to poza ifa, przed `setGameElements();`